src/app/shared/services/checkout.service.ts

```diff
@@ -49,4 +56,23 @@ export class CheckoutService {
       discount: this.discount,
     };
   }
+
+  submitAddresses(data) {
+    return new Promise((resolve, reject) => {
+      if (!data)
```

Comment:
You can extract this if condition and use it before return new Promise
On the other hand this if statement is not in any promise block (I mean block like then block), it's wrapped by main promise, so using it as return Promise.reject is not recommended